tests/stdlib/thttpclient.nim

```diff
@@ -154,8 +154,20 @@ proc ipv6Test() =
   serverFd.closeSocket()
   client.close()
 
+proc longTimeoutTest() =
+# Issue #2753
+  try:
+    var client = newHttpClient(timeout = 1000)
+    var resp = client.request("https://au.yahoo.com")
+    client.close()
+  except AssertionError:
+    doAssert false, "Exceptions should not be raised"
+  except:
+    discard
```

Comment:
What kind of test is that? "Fine as long as we get no AssertionError?!"

Reply:
Without this PR  ```doAssert``` runs.
Try exec thttpclient without this fix.

Reply:
Not my point. The test should check that either a TimeoutError is raised or nothing.

Reply:
Without try/except/this fix (and with #7813 :-)) new test outputs:
```
thttpclient.nim(177)     thttpclient
thttpclient.nim(171)     longTimeoutTest
httpclient.nim(1218)     request
httpclient.nim(1195)     request
httpclient.nim(1180)     requestAux
httpclient.nim(1073)     parseResponse
httpclient.nim(975)      parseBody
httpclient.nim(951)      parseChunks
httpclient.nim(910)      recvFull
net.nim(1221)            recv
net.nim(1199)            recv
net.nim(1170)            recv
system.nim(3846)         failedAssertImpl
system.nim(3839)         raiseAssert
system.nim(2849)         sysFatal
Error: unhandled exception: avail <= size - read  [AssertionError]
``

Reply:
Oh! In release mode sometimes:
```Error: unhandled exception: Server terminated connection prematurely [ProtocolError]```

sometimes:
```SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Segmentation fault (core dumped)```

Reply:
Well I won't add wrong tests or tests that don't work yet.